tests/test_monitor.py

```diff
@@ -759,16 +759,19 @@ def test_run_once(self):
         eq_(set([active, eternal]), remaining)
 
     def test_reap_patrons(self):
-        m = PatronReaper(self._db)
+        m = PatronRecordReaper(self._db)
         expired = self._patron()
+        credential = self._credential(patron=expired)
```

Comment:
This verifies that the one-at-a-time strategy works. The old strategy would have choked trying to delete the patron record without first deleting this Credential.